src/main/java/step1/StringAddCalculator.java

```diff
@@ -0,0 +1,112 @@
+package step1;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final String CHECK_CUSTOM_REGEX = "^//.*\\n.*$";
+    private static final String EXTRACT_DELIMITER_REGEX = "^//(.*?)\\n.*$";
+    private static final String EXTRACT_NUMS_REGEX = "^//.*\\n(.*?)$";
+    private static final String SPLIT_REGEX = ",|:";
+
+    private static String extract(String str, String regex) {
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(str);
+        if (matcher.find()) {
+            return matcher.group(1);
+        }
+        return null;
+    }
+
+    private static String extractCustomDelimiter(String str) {
+        return extract(str, EXTRACT_DELIMITER_REGEX);
+    }
+
+    private static String extractNums(String str) {
+        return extract(str, EXTRACT_NUMS_REGEX);
+    }
+
+    private static boolean checkCustomDelimiter(String str) {
+        return str.matches(CHECK_CUSTOM_REGEX);
+    }
+
+    private static boolean checkNull(String str) {
+        return str == null;
+    }
+
+    private static boolean checkEmpty(String str) {
+        return str.isEmpty();
+    }
+
+    private static boolean checkZero(String str) {
+        return checkNull(str) || checkEmpty(str);
+    }
+
+    private static String addDelimiter(String regex, String delimiter) {
+        String or = "|";
+        return regex + or + delimiter;
+    }
+
+    private static String[] splitCustomStr(String str) {
+        String delimiter = extractCustomDelimiter(str);
+        String customRegex = addDelimiter(SPLIT_REGEX, delimiter);
+        str = extractNums(str);
+        return str.split(customRegex);
+    }
+
+    private static String[] splitStr(String str) {
+        if (checkCustomDelimiter(str)) {
+            return splitCustomStr(str);
+        }
+        return str.split(SPLIT_REGEX);
+    }
+
+    private static void validateNum(int num) {
+        if (num < 0) {
+            throw new RuntimeException();
+        }
+    }
+
+    private static int parseStr(String str) {
+        int num = Integer.parseInt(str);
+        validateNum(num);
+        return Integer.parseInt(str);
```

Comment:
parse가 중복해서 되고 있네요 :) 중복을 제거하면 좋을 것 같아요!
```suggestion
        int num = Integer.parseInt(str);
        validateNum(num);
        return num;
```

Reply:
헙 원래는 `return num;` 을 하려는게 제 의도였는데 잘 짚어주셨네요. 피드백 감사합니다.